### ISSUE TITLE
Types of thunderstorms/tornadoes and different frontal/pressure systems

### ISSUE THREAD

**Issue by GGlitchX** [Mrbt0907/Weather2-Remastered#64]
This will be fairly similar to what someone else has done with their suggestion, however I would like to simplify and extend on some topics. First of all, we need to talk about different thunderstorm and tornado types, then different frontal systems. I'll also touch on pressure/ pressure systems that can trigger weather events. And I'll also talk about some unrealistic things that should be fixed.

1. Rain Showers
Already in the game and this is an improved version, however all cells should start out as a shower and develop into actual storm cells as right now they don't develop and thunderstorms come out of nowhere. There should be 3 severities of rain showers.

S1: Some light rain and **extremely** small chance of landspouts/waterspouts, very small hail and slightly damaging winds.
S2: Some slightly more heavier rain and slight chance of small hail, and extremely low chance of landspouts/waterspouts, some slightly damaging winds sometimes.
S3: Heavy rain, slight chance of weak damaging winds and small hail. Landspouts/waterspouts still very rare.

2. Single Cell
The weakest type of t-storm cell Already in the game, but this will have more improvements. These are weak pop-up storm cells that do not cause much threat other than a very low risk of landspouts and waterspouts, very small hail and possibly some slightly damaging wind. There should be 3 different severities for single cells. A storm can change through its severity stages.

S1: Light rain and slight chance of small hail, and extremely low chance of landspouts/waterspouts, some very  slightly damaging winds sometimes.
S2: Moderate rain and slight chance of hail and slight damaging winds. Very slight chance of landspouts/waterspouts.
S3: Heavy rain and moderate chance of hail and some moderate damaging winds. Very slight chance of landspouts/waterspouts.

3. Multicell
The multicell is a cluster of single cells, all fused together in one bigger cell. These are slightly more powerful storms, but still generally weak. There should be 3 severity stages. A storm can change through its severity stages.

S1: Moderate rain and slight chance of small hail. Slightly damaging winds. Low chance of landspouts/waterspouts
S2: Heavy rain, chance of small hail. Moderate damaging winds. Low chance of landspouts/waterspouts
S3: Very heavy rain, chance of small hail. Moderate damaging winds. Slight chance of landspouts/waterspouts.

4. Supercell
Already in the game. Will be revamped. Supercells are rotating thunderstorms and are formed from other t-storm types. These can produce powerful tornadoes, large hail and very damaging winds.

S1: Heavy rain, chance of moderate hail. Moderate damaging winds. Low chance of landspouts/waterspouts, and slight chance of a regular tornado/tornadic waterspout.
S2: Very heavy rain. Chance of large hail. Very damaging winds. Low chance of landspouts/waterspouts.  Slight chance of a regular tornado/tornadic waterspout.
S3: Extremely heavy rain. Chance of very large hail. Extremely damaging winds. Low chance of landspouts/waterspouts. Slight chance of a tornado/tornadic waterspout. (ps, normal supercells should be able to make tornadoes, not just hailing ones, however, hailing ones can increase the tornado risk, possibly.)

5. Rain band
Not a thunderstorm/cell, but rain bands should just be loong bands of rain with no purpose other than reducing visibility.

Extension:
Tornado strength will be calculated by it's stage. Can be customized but defaults:

Supercell S1: F-1 (7% chance) EF-2 (1% chance) EF-3 (0.25% chance) EF-4 (0.1% chance) EF-5 (0.05% chance)

Supercell S2:  EF-1 (25% chance) EF-2 (10% chance) EF-3 (2% chance) EF-4 (0.5% chance) EF-5 (0.1% chance)

Supercell S3:  EF-1 (50% chance) EF-2 (30% chance) EF-3 (10% chance) EF-4 (4% chance) EF-5 (1% chance)

Singlecell S1: (landspout/waterspout) EF-0 (2% chance) EF-1 (0.1% chance) EF-2 (0.05% chance) (EF-3 0.01% chance)
Singlecell S2: (landspout/waterspout) EF-0 (7% chance) EF1 (3% chance) EF-2 (1% chance) (EF-3 0.25% chance)
Singlecell S3: (landspout/waterspout) EF-0 (15% chance) EF-1 (8% chance) EF-2 (3% chance) EF-3 (2% chance)

Types of tornadoes:
Landspout - Weak tornado formed by low level wind shear. Goes up to EF-3.
Non-tornadic waterspout - Weak waterspout formed by low level wind shear over water. Goes up to EF-3.
Tornadic waterspout - Original tornado formed over water.

Storm components:
1. CAPE
Higher CAPE values should indicate the rarity of storm cells.
2. Lift
Indicates how strong the severity of a storm would be. More lift = stronger storms.
2. Low-level wind shear
Indicates rarity of landspout/waterspout. More low level wind shear = more landspouts/waterspouts in a thunderstorm.
3. High-level wind shear. More wind shear  = the higher the supercell severity & more supercells.
4. Pressure
Pressure will indicate if it's sunny or stormy, Low pressure = stormy weather & high pressure = sunny skies.
If there's a lot of very low pressure in an area over water, a low pressure system is formed and is marked with a red L on radar. The more pressure there is, the more the other components are boosted as well. This is vice versa for high pressure. Hurricanes/typhoons can form too if pressure is very low over water. Hurricanes will actually be nothing, but rather 
powerful storms will form in hurricanes (using all other measurements, but boost all the measurements by A LOT more) in the radius of the hurricane  and spin around the low pressure.

Fronts:
Cold fronts & warm fronts can form in low pressure systems and will boost storm components more than normal in the rest of the low pressure systems. That's it. They are very long, however.

Things to point up:
Waterspouts that are not tornadic will turn into a landspout on land, and tornadic ones will turn into a normal tornado on land.

**Comment by ProfessorFartsalot** [comment]
This looks similar to #42 . Might be interesting to merge both issues' ideas.

**Comment by Mrbt0907** [comment]
I'll take a closer look to this when I start messing with storm logic again.

This will be needed when I start making sub-weather objects and objects that hold weather objects. Once I do this, then I can start simulating fronts and weather patterns better.

**Comment by GGlitchX** [comment]
> This looks similar to #42 . Might be interesting to merge both issues' ideas.

Yes, I did mention that at the beginning of the post. This is a continuation of that, I'm basically just explaining how storms would form, and I think that person's post is very good as well.
